packages/kotti-ui/source/kotti-field-file-upload/components/DropArea.vue

```diff
@@ -0,0 +1,265 @@
+		:class="classes"
+		:data-test="dataTest ? `${dataTest}.dropArea` : undefined"
+		:tabIndex="_tabIndex"
+		@click.prevent="onClick"
+		@dragleave.prevent.stop="isDragging = false"
+		@dragover.prevent.stop="isDragging = true"
+		@drop.prevent.stop="onDrop"
+		@mouseenter.prevent.stop="isHovering = true"
+		@mouseleave.prevent.stop="isHovering = false"
+	>
+		<div class="kt-field-file-upload__drop-area__description">
+			<i class="yoco" v-text="_icon" />
+			<a v-text="translations.text.clickToUpload" />
+			{{ translations.text.dragAndDrop }}
+		</div>
+		<div
+			v-if="showInformation"
+			class="kt-field-file-upload__drop-area__information"
+		>
+			{{ informationText }}
+			<a
+				v-if="infoUrl"
+				:href="infoUrl"
+				rel="noopener noreferrer"
+				target="_blank"
+				@click.stop
+				v-text="translations.text.learnMore"
+			/>
+		</div>
+		<input
+			v-show="false"
+			ref="uploadInputRef"
+			:accept="accept"
+			:data-test="dataTest ? `${dataTest}.input` : undefined"
+			:multiple="allowMultiple"
+			type="file"
+			@change="onSelect"
+			@click.stop
+		/>
+		<div
+			v-if="$slots.footer"
+			class="kt-field-file-upload__drop-area__footer-slot"
+			@mouseenter.prevent.stop="isHovering = false"
+			@mouseleave.prevent.stop="isHovering = true"
+		>
+			<slot name="footer" />
+		</div>
+	</div>
+</template>
+
+<script lang="ts">
+import { Yoco } from '@3yourmind/yoco'
+import {
+	computed,
+	defineComponent,
+	onBeforeMount,
+	onUnmounted,
+	ref,
+} from '@vue/composition-api'
+
+import { useTranslationNamespace } from '../../kotti-i18n/hooks'
+import { KEY_ENTER, KEY_SPACE } from '../constants'
+import { KottiFieldFileUpload } from '../types'
+import { formatAccept, formatFileSize, getFileItem } from '../utils'
+import { validateSelection } from '../validators'
+
+export default defineComponent({
+	name: 'DropArea',
+	props: {
+		accept: { default: null, type: String },
+		acceptLabel: { default: null, type: String },
+		allowMultiple: { default: true, type: Boolean },
+		dataTest: { default: null, type: String },
+		icon: { default: null, type: String },
+		infoUrl: { default: null, type: String },
+		isDisabled: { default: false, type: Boolean },
+		maxFileSize: { default: null, type: Number },
+		tabIndex: { default: 0, type: Number },
+	},
+	setup(
+		props: Pick<
+			KottiFieldFileUpload.PropsInternal,
+			| 'accept'
+			| 'acceptLabel'
+			| 'allowMultiple'
+			| 'dataTest'
+			| 'icon'
+			| 'infoUrl'
+			| 'isDisabled'
+			| 'maxFileSize'
+			| 'tabIndex'
+		>,
+		{ emit },
+	) {
+		const translations = useTranslationNamespace('KtFieldFileUpload')
+
+		const dropAreaRef = ref<HTMLDivElement | null>(null)
+		const isDragging = ref<boolean>(false)
+		const isHovering = ref<boolean>(false)
+		const uploadInputRef = ref<HTMLInputElement | null>(null)
+
+		const emitFiles = (files: FileList | null) => {
+			if (props.isDisabled || !files || files.length === 0) return
+
+			const { accept, allowMultiple, maxFileSize } = props
```

Comment:
props is a `reactive`

IF (but since it's not callled useValidateSelection, I'll asume it's a NO) you rely on reactivity of those props, you'd have just destroyed reactivity

Reply:
I'd say regardless - change it, that would be a nasty bug to track down again